sql/memo/join_order_builder.go

```diff
@@ -892,28 +890,30 @@ type edge struct {
 	rules []conflictRule
 }
 
-func (e *edge) populateEdgeProps(allV vertexSet, vertexes []GroupId, edges []edge) {
+func (e *edge) populateEdgeProps(tableIds []sql.TableId, edges []edge) {
 	var tables sql.FastIntSet
 	var cols sql.ColSet
-	var nullAccepting []ScalarExpr
+	var nullAccepting []sql.Expression
 	if len(e.filters) > 0 {
 		for _, e := range e.filters {
-			sProps := e.Group().ScalarProps()
-			cols = cols.Union(sProps.Cols)
-			tables = tables.Union(sProps.Tables)
-			if !sProps.nullRejecting {
+			eCols, eTabs, nullRej := getExprScalarProps(e)
+			cols = cols.Union(eCols)
+			tables = tables.Union(eTabs)
+			if !nullRej {
```

Comment:
It's odd that sometimes you use `nullAccepting` and other times `nullRejecting`. Should pick one and use it everywhere

Reply:
this is unused right now, can just delete

Reply:
We've talked about this before, but academically everything is in terms of null-rejecting properties. But practically sometimes we need a list of the filters that violate that property